nodejs8 function is not deploying with private repo reference

**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT
**What happened**:
I was trying to deploy a `nodejs8 http function` with a `dependency` file and few `environment variables`. One among the environment variable is setting the `private npm repo url`. When I ran the deploy command, it started deploying by accepting all the flags. But during the `Install container` phase, it goes under **crashloop** with the following error.

```
npm ERR! path /.npmrc
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall open
npm ERR! Error: EACCES: permission denied, open '/.npmrc'
npm ERR!  { Error: EACCES: permission denied, open '/.npmrc'
npm ERR!   stack: 'Error: EACCES: permission denied, open \'/.npmrc\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'open',
npm ERR!   path: '/.npmrc' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

┌──────────────────────────────────────────────┐
│           npm update check failed            │
│     Try running with sudo or get access      │
│     to the local update config store via     │
│ sudo chown -R $USER:$(id -gn $USER) /.config │
└──────────────────────────────────────────────┘
```

My **deploy command** looks like the below,

```
kubeless function deploy elastic-writer --runtime nodejs8 \
					--handler elastic-writer.writer \
					--from-file elastic-writer.js \
					--dependencies package.json \
					--env http_proxy:<http-proxy>  \
					--env https_proxy:<https-proxy> \
					--env HTTPS_PROXY:<https-proxy> \
					--env HTTP_PROXY:<http-proxy> \
					--env no_proxy:<no-proxies> \
					--env NPM_REGISTRY=<internal-private-repo> \
					--env NPM_SCOPE=@internal
```

**What you expected to happen**:
To set the private repository in the .npmrc and run seamlessly.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy a nodejs function with a private repository configuration passed as the environmental variable.

**Anything else we need to know?**:
I guess, the `install` container doesn't have enough permissions. Attached the contents of the `kubectl describe po elastic-writer-2651850065-qpbdf` command below.
[events.txt](https://github.com/kubeless/kubeless/files/1905744/events.txt)

**Environment**:
 ```
 oc v1.5.1+7b451fc
 kubernetes v1.5.2+43a9be4
 features: Basic-Auth

 Server <Internal-Cluster>
 openshift v3.7.9
 kubernetes v1.7.6
```

- Kubernetes version (use `kubectl version`): 1.7
- Kubeless version (use `kubeless version`): v0.5.0
- Cloud provider or physical cluster: OpenShift v3.7

--- a/docker/runtime/golang/Dockerfile.init
+++ b/docker/runtime/golang/Dockerfile.init
@@ -3,13 +3,12 @@ FROM golang:1.10
 # Install dep
 RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
 RUN go get golang.org/x/net/context
-RUN mkdir -p /go/pkg/dep && chown 1000:1000 /go/pkg/dep
+RUN mkdir -p /go/pkg/dep && chmod -R a+w /go/pkg/dep
 
 # Prepare function environment
 RUN ln -s /kubeless $GOPATH/src/kubeless 
 
 # Install controller
-USER 1000
 RUN mkdir -p $GOPATH/src/github.com/kubeless/kubeless/pkg/functions
 ADD pkg/functions/* $GOPATH/src/github.com/kubeless/kubeless/pkg/functions
 RUN mkdir -p $GOPATH/src/controller
@@ -17,3 +16,7 @@ ADD docker/runtime/golang/Gopkg.toml $GOPATH/src/controller/
 WORKDIR $GOPATH/src/controller/
 ADD docker/runtime/golang/kubeless.tpl.go $GOPATH/src/controller/
 RUN dep ensure
+RUN chmod -R a+w $GOPATH
+RUN chmod -R a+w /go/pkg/dep
+
+USER 1000

--- a/docker/runtime/golang/Dockerfile.init
+++ b/docker/runtime/golang/Dockerfile.init
@@ -3,7 +3,7 @@ FROM golang:1.10
 # Install dep
 RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
 RUN go get golang.org/x/net/context
-RUN mkdir -p /go/pkg/dep && chmod -R a+w /go/pkg/dep
+RUN mkdir -p /go/pkg/dep
 
 # Prepare function environment
 RUN ln -s /kubeless $GOPATH/src/kubeless 

--- a/docker/runtime/php/Dockerfile
+++ b/docker/runtime/php/Dockerfile
@@ -48,8 +48,7 @@ COPY config/default.conf /etc/nginx/conf.d/default.conf
 RUN rm /etc/nginx/sites-available/default && \
     mkdir -p /app && \
     touch /run/nginx.pid && \
-    chown 1000:1000 -R /run/nginx.pid /app /var/lib/nginx /var/log/nginx
+    chmod -R a+w /run/nginx.pid /app /var/lib/nginx /var/log/nginx
 USER 1000
 
-ENTRYPOINT [ "" ]
 CMD ["/usr/bin/supervisord", "-c", "/etc/supervisor/conf.d/supervisord.conf"]

--- a/kubeless-non-rbac.jsonnet
+++ b/kubeless-non-rbac.jsonnet
@@ -140,7 +140,7 @@ local runtime_images ='[
       {
         "name": "php72",
         "version": "7.2",
-        "runtimeImage": "kubeless/php@sha256:82b94c691302bc82f3900444255cabb8f230487764eafeba7866ac49d90ddc3b",
+        "runtimeImage": "kubeless/php@sha256:4e44ab60f597e93097bf9f5ea91d58bd9c308bf206043db2a9809ec16a8ff2f4",
         "initImage": "composer:1.6"
       }
     ],
@@ -155,7 +155,7 @@ local runtime_images ='[
         "name": "go1.10",
         "version": "1.10",
         "runtimeImage": "kubeless/go@sha256:bf72622344a54e4360f31d3fea5eb9dca2c96fbedc6f0ad7c54f3eb8fb7bd353",
-        "initImage": "kubeless/go-init@sha256:ce6ef4fafe518ed78b3a68b03947c064fec1cf8c667cd109e9331f227877b3a9"
+        "initImage": "kubeless/go-init@sha256:e262f70639594b3a9e3481843171ecbbe82e84b786825ebe28bc1a3ae89310d3"
       }
     ],
     "depName": "Gopkg.toml",

--- a/kubeless-openshift.jsonnet
+++ b/kubeless-openshift.jsonnet
@@ -1,10 +1,14 @@
 # Builds on kubeless.ksonnet to produce a deployable manifest on OpenShift 1.5
 # Modifies apiVersion for kubeless-controller Deployment to extensions/v1beta1
 # Modifies ClusterRole and ClusterRoleBinding apiVersions to v1
+local k = import "ksonnet.beta.1/k.libsonnet";
 local kubeless = import "kubeless.jsonnet";
+
+local config = kubeless.cfg + k.core.v1.configMap.data({"deployment":'{"spec":{"template":{"spec":{"securityContext":{}}}}}'});
 
 kubeless + {
   controller: kubeless.controller + { apiVersion: "extensions/v1beta1" },
   controllerClusterRole: kubeless.controllerClusterRole + { apiVersion: "v1" },
   controllerClusterRoleBinding: kubeless.controllerClusterRoleBinding + { apiVersion: "v1" },
+  cfg: config,
 }

--- a/pkg/langruntime/langruntime.go
+++ b/pkg/langruntime/langruntime.go
@@ -210,6 +210,8 @@ func (l *Langruntimes) GetBuildContainer(runtime, depsChecksum string, env []v1.
 	case strings.Contains(runtime, "nodejs"):
 		registry := "https://registry.npmjs.org"
 		scope := ""
+		// Force HOME to a folder with permissions to avoid issues in OpenShift #694
+		env = append(env, v1.EnvVar{Name: "HOME", Value: "/tmp"})
 		for _, v := range env {
 			if v.Name == "NPM_REGISTRY" {
 				registry = v.Value

--- a/pkg/langruntime/langruntime_test.go
+++ b/pkg/langruntime/langruntime_test.go
@@ -137,6 +137,15 @@ func TestGetBuildContainer(t *testing.T) {
 	if !strings.Contains(c.Args[0], "npm config set myorg:registry http://reg.com && npm install") {
 		t.Errorf("Unexpected command %s", c.Args[0])
 	}
+	home := v1.EnvVar{Name: "HOME"}
+	for _, v := range c.Env {
+		if v.Name == "HOME" {
+			home = v
+		}
+	}
+	if home.Value != "/tmp" {
+		t.Error("It should set /tmp as home")
+	}
 
 	// It should return the proper build image for ruby
 	c, err = lr.GetBuildContainer("ruby2.4", "abc123", env, vol1)

